include/watchdog.h

```diff
@@ -55,62 +70,237 @@ struct wdt_config {
+	void (*get_config)(struct device *dev, struct wdt_config *config);
+	int (*set_config)(struct device *dev, struct wdt_config *config);
+	void (*reload)(struct device *dev);
 };
+/**
+ * @endcond
+ */
 
-static inline void wdt_enable(struct device *dev)
+/**
+ * @brief Set up watchdog instance.
+ *
+ * This function is used for configuring global watchdog settings that
+ * affect all timeouts. It should be called after installing timeouts.
+ * After successful return all installed timeouts are valid.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param options Bit field with following parts:
+ *
+ *     pause in sleep      [ 0 ]       - pause in CPU sleep
+ *     pause by debugger   [ 1 ]       - pause when halted by debugger
+ *
+ * @retval 0 If success.
+ * @retval -ENOTSUP If any of the set options is not supported.
+ * @retval -EBUSY If watchdog instance has been already setup.
+ */
+static inline int wdt_setup(struct device *dev, u8_t options)
+{
+	const struct wdt_driver_api *api = dev->driver_api;
+
+	return api->setup(dev, cfg);
+}
+
+/**
+ * @brief Disable watchdog instance.
+ *
+ * This function stops watchdog instance and automatically uninstalls all
+ * timeouts. To set up watchdog again it is required to install timeouts and
+ * call wdt_setup function. Not all watchdogs can be re-started once they are
+ * disabled.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @retval 0 If success.
+ * @retval -EFAULT If watchdog instance is not enabled.
+ * @retval -EPERM If watchdog can not be disabled directly by application code.
+ */
+static inline int wdt_disable(struct device *dev)
+{
+	const struct wdt_driver_api *api = dev->driver_api;
+
+	return api->disable(dev);
+}
+
+/**
+ * @brief Install new timeout.
+ *
+ * This function should be used before wdt_setup(). Changes applied here
+ * have no effects until wdt_setup() is called.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param cfg Pointer to timeout configuration structure.
+ *
+ * @retval 0 If success.
+ * @retval -EBUSY If timeout can not be installed while watchdog has already
+ *		  been setup.
+ * @retval -ENOMEM If no more timeouts can be installed.
+ * @retval -EFAULT If timeout with given handle has already been installed.
+ *		   Applies to multi-channel watchdogs only.
+ * @retval -ENOTSUP If any of the set flags is not supported.
+ * @retval -EINVAL If any of the window timeout value is out of possible range.
+ *		   This value is also returned if watchdog supports only one
+ *		   timeout value for all timeouts and supplied timeout window
+ *		   differs from windows for alarms installed so far.
+ */
+static inline int wdt_install_timeout(struct device *dev,
+				      struct wdt_timeout_cfg *cfg)
+{
+	const struct wdt_driver_api *api = dev->driver_api;
+
+	return api->install_timeout(dev, cfg);
+}
+
+/**
+ * @brief Reload specified watchdog timeout.
```

Comment:
"Feed specified watchdog timeout." would sound strange, but using 'reload' interchangeably with 'feed' may also be non obvious. Wouldn't 'service' be a better choice. As in "Service specified watchdog timeout."?

Reply:
But the function name is wdt_feed so "feed" is consistent.  Should the function name be wdt_service?

Reply:
So let's stay with "feed". Thanks @dbkinder for the review!